sim/battle.js

```diff
@@ -2678,6 +2678,7 @@ class Battle extends Dex.ModdedDex {
 
 		if (avatar) player.avatar = avatar;
 		this.add('player', player.id, player.name, avatar);
+		this.add('teamsize', player.id, player.pokemon.length);
```

Comment:
May involve some refactoring, though. I guess it's fine like this until I personally refactor it later...